### DISCUSSION TITLE
RabbitMQ 3.13 will support MQTT 5.0

### DISCUSSION THREAD

**Discussion by ansd** [rabbitmq/rabbitmq-server#9539] [Other]
New blog post from Team RabbitMQ: [MQTT 5.0 support is coming in RabbitMQ 3.13](https://blog.rabbitmq.com/posts/2023/07/mqtt5/).

You can try 3.13 today on your specific workloads using [a beta release](https://github.com/rabbitmq/rabbitmq-server/discussions/9313).

**Comment by petersilva** [comment]
For clarity's sake: it would be more accurate to say that **some** or  **a subset** of MQTT 5.0 is coming to RabbitMQ 3.13 ... **no shared subscriptions** makes it useless to me.

**Reply by michaelklishin** [reply]
@petersilva the post makes it very clear that there are limitations. 

It's not uncommon to see MQTT 5 implementations to implement most but not all features, and still use the term "MQTT 5" and not "a subset of MQTT 5".

I do feel that Team RabbitMQ is being clear and honest about what we are announcing here.

**Reply by petersilva** [reply]
Which implementations are you referring to, and what did those others leave out?  I have looked into: mosquitto, EMQX, HiveMQ brokers, which claim to support MQTTv5, and do support shared subscriptions, and as far as I can tell there are no other omissions from standards support. All the paho clients seem to be the same. They don't seem to be omitting anything.

I agree Rabbitmq's release notes are thorough and clear, and the omission is documented, but you have to read the fine print.  I don't see evidence of any similar omissions in any of the other mqtt v5 implementations that I am aware of, and I haven't run into any constraints or limitations in my limited experience with them. As far as I can tell, they are complete mqtt v5 implementations.  For my usage omission of shared subscriptions means I can't use it at all... so I don't feel v5 support is really there.

Shared subscriptions is one of the headline features of MQTTv5.  The omission is material.

On Amazon's IOT Core product, blurb, shared subscriptions are the first feature they list.

* https://docs.aws.amazon.com/iot/latest/developerguide/mqtt.html#mqtt5

**Reply by michaelklishin** [reply]
@petersilva well I'm sorry that our MQTTv5 is "useless" to you. Many folks are happy to see MQTT support in RabbitMQ progress forward at the pace.

Our team is small and we support N protocols. Support for shared subscriptions in MQTTv5 will come at some point. 

Partial support for protocols even for the largest vendors is something that is fairly common in the industry, for all kinds of reasons. I'm not interested in having a finger pointing contest.

**Comment by michaelklishin** [comment]
Demands for more features and claims that this work is "useless" will not change the scope of 3.13. We have noted some interest in support for shared subscriptions in MQTTv5 (not that it was news to us) => locking this as resolved.
